libecole/tests/src/observation/test-pseudocosts.cpp

```diff
@@ -0,0 +1,59 @@
+#include <limits>
+#include <memory>
+#include <stdexcept>
+#include <tuple>
+
+#include <catch2/catch.hpp>
+
+#include "ecole/environment/branching.hpp"
+#include "ecole/environment/exception.hpp"
+#include "ecole/observation/pseudocosts.hpp"
+#include "ecole/reward/isdone.hpp"
+
+#include "conftest.hpp"
+
+using namespace ecole;
+
+TEST_CASE("Pseudocosts") {
```

Comment:
So unittests for observations are a bit of a mess right now, and I'll work on making them better.
Those tests here show that the interface works correctly with Ecole.

I'd like to see a functional test, something that says, "_Hey look, my function is not returning a vector of random number, this is a correct output_".
I understand it is hard to predict what the pseudo cost will be, so maybe tests that

- The vector has as many entries as there are variables;
- The vector is not filled only with `nan`
- Pseudo cost are positive (for minimization problems but not sure of SCIP conventions here)
- If a variable is fractional, it has a pseudo cost
- ...